src/EA.Weee.Web.Tests.Unit/Areas/Admin/Controllers/ReportsControllerTests.cs

```diff
@@ -1816,5 +1819,39 @@ public async Task PostAatfAeReturnData_Always_SetsInternalBreadcrumbToViewReport
             // Assert
             Assert.Equal("View reports", breadcrumb.InternalActivity);
         }
+
+        [Fact]
+        public async void PostAatfAeReturnData_OnDownload_SetsURL()
+        {
+            IWeeeClient apiClient = A.Fake<IWeeeClient>();
+            ReportsController controller = new ReportsController(
+                () => apiClient,
+                A.Dummy<BreadcrumbService>());
+
+            var httpContext = new HttpContextMocker();
+            httpContext.AttachToController(controller);
+            var httpRequest = A.Fake<HttpRequestBase>();
+
+            CSVFileData file = new CSVFileData() { FileContent = "Content", FileName = "test.csv" };
+
+            Uri uri = new Uri("https://localhost:44300/admin/");
+            A.CallTo(() => controller.HttpContext.Request).Returns(httpRequest);
+            A.CallTo(() => controller.HttpContext.Request.Url).Returns(uri);
+
+            A.CallTo(() => apiClient.SendAsync(A<string>._, A<GetAatfAeReturnDataCsv>._)).Returns(file);
+
+            ActionResult result = await controller.DownloadAatfAeDataCsv(2019, 1,
+                FacilityType.Aatf, A.Dummy<int?>(), A.Dummy<Guid?>(), A.Dummy<Guid?>(), 
+                A.Dummy<Guid?>());
+
+            A.CallTo(() => apiClient.SendAsync(A<string>._, A<GetAatfAeReturnDataCsv>.Ignored))
+                .MustHaveHappened(Repeated.Exactly.Once);
+
+            A.CallTo(() => apiClient.SendAsync(A<string>._, A<GetAatfAeReturnDataCsv>.That.Matches(x => x.AatfDataUrl.Equals("https://localhost:44300/admin/aatf/details/"))));
```

Comment:
Need to add .MustHaveHappend on this check

Reply:
Added